Generate.py

```diff
@@ -256,18 +333,217 @@ def get_choice_legacy(option, root, value=None) -> Any:
+                val_str = s[i:j]
+                if len(val_str.split(".")) > 1:
+                    val = float(val_str)
+                else:
+                    val = int(val_str)
+                tokens.append(NumberToken(val))
+                i += len(val_str)
+                state = "end"
+            elif s[i].isalpha():
+                j = i + 1
+                while j < len(s) and (s[j].isalpha() or s[j].isdigit() or s[j] == "_"):
+                    j += 1
+                val_str = s[i:j]
+                tokens.append(NameToken(val_str))
+                i += len(val_str)
+                state = "end"
+            else:
+                raise ValueError(f"unexpected symbol {s[i]} at {i} in: {s}")
+        else:
+            assert assert_type(state, Literal["end"]) == "end", f"{state=}"
+            if s[i] == ")":
+                if paren_count <= 0:
+                    raise ValueError(f"unmatched close parentheses at {i} in: {s}")
+                paren_count -= 1
+                tokens.append(CloseParenToken(")"))
+                i += 1
+            elif _is_bin_op(val_str := s[i]):
+                tokens.append(BinOpToken(val_str))
+                state = "start"
+                i += 1
+            else:
+                raise ValueError(f"unexpected symbol {s[i]} at {i} in: {s}")
+    if paren_count != 0:
+        raise ValueError(f"unclosed parentheses in: {s}")
+    if state != "end":
+        raise ValueError(f"invalid expression: {s}")
+    return tokens
+
+
+def infix_to_postfix(infix: Sequence[AllTokens]) -> Sequence[PostfixTokens]:
+    infix = [OpenParenToken("(")] + list(infix) + [CloseParenToken(")")]
+    op_stack: List[Union[OpenParenToken, BinOpToken]] = []
+    postfix: List[PostfixTokens] = []
+    for token in infix:
+        if isinstance(token, OpenParenToken):
+            op_stack.append(token)
+        elif isinstance(token, CloseParenToken):
+            while len(op_stack):
+                op = op_stack.pop()
+                if isinstance(op, OpenParenToken):
+                    break
+                else:
+                    assert isinstance(assert_type(op, BinOpToken), BinOpToken), f"{op=}"
+                    postfix.append(op)
+        elif isinstance(token, BinOpToken):
+            while True:
+                assert len(op_stack), f"{infix=}"
+                top = op_stack[-1]
+                if _precedence[top.val] >= _precedence[token.val]:
+                    top = op_stack.pop()
+                    assert isinstance(top, BinOpToken), f"{top=} {infix=}"
+                    postfix.append(top)
+                else:
+                    break
+            op_stack.append(token)
+        else:
+            assert isinstance(token, NumberToken) or isinstance(assert_type(token, NameToken), NameToken), f"{token=}"
+            postfix.append(token)
+    return postfix
+
+
+def eval_postfix(postfix: Sequence[PostfixTokens], root: dict, record: ChoiceRecord) -> float:
```

Comment:
https://github.com/Vertraic/Archipelago/pull/1

I think it's better if the math parsing is separate from any `Option` choosing stuff.